From e17d2b458d5ed936d316f99ad04d7448cce75eea Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 24 Aug 2018 12:09:19 +0300
Subject: [PATCH] [post-image.sh] SPI Errata was fixed for Gaudi, update code

Now no need to do swap of SPL (copied by preboot from
SPI memory mapped address)
---

diff --git a/board/habanalabs/common/post-image.sh b/board/habanalabs/common/post-image.sh
index 8e4e7aa..1798912 100755
--- a/board/habanalabs/common/post-image.sh
+++ b/board/habanalabs/common/post-image.sh
@@ -12,10 +12,15 @@
 	INF=$1
 	OUTF=$2
 
-	# SWAP needs this to be 4B aligned in size
-	dd status=none conv=sync bs=4 if=${INF} > ${OUTF}.tmp
-	# Sadly --endian=big is not swapping for -tx1 so I used some gawk magic
-	od -Anone -v -tx1 -w4 ${OUTF}.tmp | gawk '{ print $4 $3 $2 $1 }' | xxd -r -p > ${OUTF}
+	if [ "${HL_DEV}" == "dali" ]; then
+		# SWAP needs this to be 4B aligned in size
+		dd status=none conv=sync bs=4 if=${INF} > ${OUTF}.tmp
+		# Sadly --endian=big is not swapping for -tx1 so I used some gawk magic
+		od -Anone -v -tx1 -w4 ${OUTF}.tmp | gawk '{ print $4 $3 $2 $1 }' | xxd -r -p > ${OUTF}
+	else
+		# No swap is needed, errata was fixed
+		dd status=none conv=sync bs=4 if=${INF} > ${OUTF}
+	fi
 }
 
 pushd ${BINARIES_DIR}
